reference-implementation/to-upstream-wpts/writable-streams/start.js

```diff
@@ -15,17 +15,11 @@ async_test(t => {
       });
     },
     write(chunk) {
-      t.step(() => {
-        assert_true(expectWriteCall, 'write should not be called until start promise resolves');
-        assert_equals(chunk, 'a', 'chunk should be the value passed to write');
-        t.done();
-      });
+      assert_true(expectWriteCall, 'write should not be called until start promise resolves');
+      assert_equals(chunk, 'a', 'chunk should be the value passed to write');
```

Comment:
I think this should be a promise_test, but you can use the recordingWritableStream to test that it gets passed the right arguments. Example at https://github.com/whatwg/streams/blob/ba1e784ec2761fa7e992c80e1c3ef0f10b991ce1/reference-implementation/to-upstream-wpts/writable-streams/bad-underlying-sinks.js#L74-L76

That will also take care of the asserting that close is not called.

In general I think many of the tests in this file that do asserts that certain underlying sink methods are called would benefit from this.